src/test/java/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java

```diff
@@ -140,32 +130,32 @@ public void testSpecificAttachments() {
       sb.append(validateAttachment(clazz));
     }
     if (sb.length() > 0) {
-      System.out.println(sb.toString());
-      // fail(sb.toString());
+      fail("One or more attachments are invalid:\n" + sb.toString());
     }
   }
 
   /**
-   * Scans the compiled /classes folder and finds all classes that implement IAttachment to verify that
+   * Scans the compiled classes folder and finds all classes that implement IAttachment to verify that
    * all @GameProperty have valid setters and getters.
    */
   @Test
-  public void testAllAttachments() {
-    // find the classes folder
-    final URL url = getClass().getResource("/");
-    File file = null;
-    try {
-      file = new File(url.toURI());
-      file = new File(file.getParent(), "classes");
-    } catch (final URISyntaxException e) {
-      fail(e.getMessage());
-      ClientLogger.logQuietly(e);
+  public void testAllAttachments() throws Exception {
```

Comment:
This test was previously broken because it was using the wrong classpath (probably prior to the switch to Gradle).  I fixed it so that it works when run from both Eclipse and Gradle (I don't know if it works under IntelliJ).  However, it has several issues:

* It loads nearly _every_ TripleA class on the classpath.
    * It can be configured to skip certain problematic classes via the `SKIPCLASSES` field, but that smells a bit and is yet another list of strings to be maintained.
    * Because it attempts to load some UI classes, it cannot be run in a headless environment.  Thus, it is skipped when this condition is true (i.e. it won't run on Travis, where it would probably be the most useful).
* It is a relatively long-running unit test because of the previous issue (it averages about 10 seconds on my machine).

I'm not sure if it's worth keeping this test given that we can provide a whitelist of classes to test in `testSpecificAttachments()`.  The obvious drawback is if someone adds a new attachment, and forgets to update the list in `getKnownAttachmentClasses()`, the new attachment could go untested.  This has apparently happened recently, as the `UserActionAttachment` class was not added to this list when that attachment was added to the codebase.